server/src/main/java/org/opensearch/search/query/QueryCollectorContext.java

```diff
@@ -67,38 +68,52 @@
  */
 @PublicApi(since = "1.0.0")
 public abstract class QueryCollectorContext {
-    private static final Collector EMPTY_COLLECTOR = new SimpleCollector() {
-        @Override
-        public void collect(int doc) {}
 
-        @Override
-        public ScoreMode scoreMode() {
-            return ScoreMode.COMPLETE_NO_SCORES;
-        }
-    };
+    private static Collector createEmptyCollector(ScoreMode scoreMode) {
```

Comment:
Could you please reference the relevant Apache Lucene issue as we've discussed, so it will be clear why we are doing score mode "dance", thanks!